daml-lf/transaction-scalacheck/src/main/scala/com/digitalasset/daml/lf/value/TypedValueGenerators.scala

```diff
@@ -118,12 +127,17 @@ object TypedValueGenerators {
       import scalaz.std.vector._
       type Inj[Cid] = Vector[elt.Inj[Cid]]
       override val t = TypePrim(PT.List, ImmArraySeq(elt.t))
-      override def inj[Cid] = elts => ValueList(elts.map(elt.inj).to[FrontStack])
+      override def inj[Cid: IntroCtx](elts: Inj[Cid]) =
+        ValueList(elts.map(elt.inj(_)).to[FrontStack])
       override def prj[Cid] = {
         case ValueList(v) => v.toImmArray.toSeq.to[Vector] traverse elt.prj
         case _ => None
       }
-      override def injarb[Cid: Arbitrary] = {
+      override def injord[Cid: Order] = {
+        implicit val e: Order[elt.Inj[Cid]] = elt.injord
+        implicitly[Order[Vector[elt.Inj[Cid]]]]
```

Comment:
I am not very sure what implicit is infered there, but if `scalaz.std.VectorOrder` seems to take the size of the compared vectors first, and then compare lexicographically the content. It is not what we want here. Problably `scalaz.std.ListOrder` is better,